frontend/src/scenes/persons/PersonDeleteModal.tsx

```diff
@@ -1,75 +1,90 @@
+
+                <div className="space-y-2">
+                    <label className="text-sm">
+                        To confirm, please type <strong>{DELETE_CONFIRMATION_TEXT}</strong> below:
+                    </label>
+                    <LemonInput
+                        value={deleteConfirmationText}
+                        onChange={setDeleteConfirmationText}
+                        placeholder={DELETE_CONFIRMATION_TEXT}
+                        className="w-full"
+                        autoFocus
                     />
-                </>
-            }
-            footer={
-                <>
-                    <LemonButton
-                        status="danger"
-                        type="tertiary"
-                        onClick={() => deletePerson(personDeleteModal as PersonType, true)}
-                        disabled={!isDeletionConfirmed}
-                        data-attr="delete-person-with-events"
-                    >
-                        Delete person and all corresponding events
-                    </LemonButton>
-                    <LemonButton
-                        type="secondary"
-                        onClick={() => showPersonDeleteModal(null)}
-                        data-attr="delete-person-cancel"
-                    >
-                        Cancel
-                    </LemonButton>
-                    <LemonButton
-                        type="secondary"
-                        status="danger"
-                        onClick={() => deletePerson(personDeleteModal as PersonType, false)}
-                        disabled={!isDeletionConfirmed}
-                        data-attr="delete-person-no-events"
-                    >
-                        Delete person
-                    </LemonButton>
-                </>
-            }
-        />
+                </div>
+                <div className="bg-warning-highlight border border-warning rounded p-2 text-sm">
+                    Do NOT delete this person if you want to re-use the distinct IDs. Instead use split ID. Re-using the
+                    distinct ID of a deleted person is not supported and will result in a bad application state.
+                </div>
+            </div>
+            <div className="flex justify-end gap-2 mt-4">
+                <LemonButton type="secondary" onClick={handleClose}>
+                    Cancel
+                </LemonButton>
+                <LemonButton
+                    type="primary"
+                    disabledReason={
+                        deleteConfirmationText !== DELETE_CONFIRMATION_TEXT
+                            ? 'Please type the correct confirmation text'
+                            : undefined
+                    }
+                    onClick={() =>
+                        deletePerson(personDeleteModal as PersonType, alsoDeleteEvents, alsoDeleteRecordings)
+                    }
+                >
+                    Delete Person
```

Comment:
sentence case at posthog
(despite the no uppercase in socials habit :))
```suggestion
                    Delete person
```